zerver/lib/bugdown/__init__.py

```diff
@@ -281,6 +286,46 @@ def add_vimeo_preview(root: Element, link: Text, extracted_data: Dict[Text, Any]
         img = markdown.util.etree.SubElement(anchor, "img")
         img.set("src", img_link)
 
+
+def add_quoted_message_preview(sender: UserProfile, root: Element, link: Text, message_id: Text) -> None:
+    from zerver.lib.message import access_message
+
+    container = markdown.util.etree.SubElement(root, "div")
+    blockquote_element = markdown.util.etree.SubElement(container, 'blockquote')
+    element_string = '<root>' +\
```

Comment:
It is generally recommended to use the tree manipulation operations instead of generating an HTML string by hand.